app/assets/src/components/Samples.jsx

```diff
@@ -1143,8 +1102,14 @@ class Samples extends React.Component {
 
     let check_all = (
       <div className="check-all">
-        <input type="checkbox" id="checkAll" className="filled-in checkAll" />
-        <label htmlFor="checkAll" />
+          <input type="checkbox"
+            id="checkAll"
+            className="filled-in checkAll"
+            checked = {this.state.allChecked}
+            onChange = {(e) => {}}
```

Comment:
Defining a function in a component prop is generally bad practice as it will create a new function each time the component is rendered which means it will cause unnecessary re-rendering as react will always think that the onChange property changed.

Reply:
OK. I think I got a warning without having it but I will remove it and try again.